src/Raven.Studio/typescript/viewmodels/database/settings/databaseSettings.ts

```diff
@@ -0,0 +1,454 @@
+        if (entry instanceof models.databaseEntry && entry.override()) {
+            stateClass = this.isValid(entry.validationGroup) ? "customized-item" : "customized-item-with-error";
+        }
+        
+        return stateClass;
+    }
+    
+    private initSummaryGrid() {
+        const summaryGrid = this.summaryGridController();
+        summaryGrid.headerVisible(true);
+
+        summaryGrid.init(() => this.fetchSummaryData(), () =>
+            [
+                new textColumn<models.settingsEntry>(summaryGrid, x => x.keyName(), "Configuration Key", "30%", {
+                    sortable: "string",
+                    extraClass: x => this.getEntryStateClass(x)
+                }),
+                new textColumn<models.settingsEntry>(summaryGrid, x => x.effectiveValue(), "Effective Value", "40%", {
+                    sortable: "string",
+                    extraClass: (x) => x.effectiveValue() ? "effective-value" : ""
+                }),
+                new textColumn<models.settingsEntry>(summaryGrid, x => x.effectiveValueOrigin(), "Origin", "20%", {
+                    sortable: "string",
+                    extraClass: (x) => {
+                        let classes = "source-item ";
+                        switch (x.effectiveValueOrigin()) {
+                            case "Database":
+                                classes += "source-item-database";
+                                return x.effectiveValue() ? classes : classes += " null-value";
+                            case "Server":
+                                classes += "source-item-server";
+                                return x.effectiveValue() ? classes : classes += " null-value";
+                            case "Default":
+                                classes += "source-item-default";
+                                return x.effectiveValue() ? classes : classes += " null-value";
+                        }
+                    }
+                })
+            ]);
+    }
+    
+    private fetchSummaryData(): JQueryPromise<pagedResult<models.settingsEntry>> {
+        const entriesForSummaryMode = _.filter(this.allEntries(), x => x.showEntry());
+        
+        return $.when<pagedResult<models.settingsEntry>>({
+            items: entriesForSummaryMode,
+            totalResultCount: entriesForSummaryMode.length,
+            resultEtag: null,
+            additionalResultInfo: undefined
+        })
+    }
+
+    private fetchDatabaseSettings(db: database, refresh: boolean = false): JQueryPromise<Raven.Server.Config.SettingsResult> {
+        eventsCollector.default.reportEvent("database-settings", "get");
+
+        return new getDatabaseSettingsCommand(db, refresh)
+            .execute()
+            .done((result: Raven.Server.Config.SettingsResult) => {
+                
+                const settingsEntries = result.Settings.map(x => {
+                    const rawEntry = x as Raven.Server.Config.ConfigurationEntryDatabaseValue;
```

Comment:
let's move this logic to abstract class to static method?

Reply:
done